nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -112,10 +112,28 @@ let
 
       NIX_DISK_IMAGE=$(readlink -f "''${NIX_DISK_IMAGE:-${config.virtualisation.diskImage}}")
 
-      if ! test -e "$NIX_DISK_IMAGE"; then
-          ${qemu}/bin/qemu-img create -f qcow2 "$NIX_DISK_IMAGE" \
-            ${toString config.virtualisation.diskSize}M
-      fi
+      ${if cfg.buildRootImage then
+        ''
+          if ! test -e "$NIX_DISK_IMAGE"; then
+            # Create a writable copy/snapshot of the root disk.
+            ${qemu}/bin/qemu-img create -f qcow2 -F qcow2 -b ${rootImage}/nixos.qcow2 "$NIX_DISK_IMAGE"
+          elif test -z "$IGNORE_EXISTING_IMAGE"; then
+            cat >&2 <<EOF
+          WARNING: Root disk image already exists at $NIX_DISK_IMAGE!
+                   If you've rebuilt the image, you need to remove the existing
+                   disk image for the changes to apply! To ignore this warning
+                   and start immediately, set the IGNORE_EXISTING_IMAGE variable.
+          EOF
+          sleep 20
```

Comment:
Wouldn't it be a better idea to abort here if the variable is not set instead of sleeping?